# Identitas Penulis
- **Nama Lengkap**: Muhammad Ikctiar Saputra
- **NIM**: 250401020169
- **Kelas**: IF401
- **Program Studi**: PJJ Informatika

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

URL = "https://raw.githubusercontent.com/mwaskom/seaborn-data/master/penguins.csv"
df = pd.read_csv(URL)
# Menghapus baris kosong agar plotting berjalan bersih tanpa warning
df = df.dropna()
print("Dataset shape:", df.shape)

Dataset shape: (333, 7)


In [2]:
# Langkah 2: Statistik Deskriptif Lengkap
print("Deskripsi Dataset:")
print(df.describe().round(2))

Deskripsi Dataset:
       bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
count          333.00         333.00             333.00       333.00
mean            44.02          17.16             200.97      4207.06
std              5.47           1.97              14.02       805.22
min             32.10          13.10             172.00      2700.00
25%             39.50          15.60             190.00      3550.00
50%             44.50          17.30             197.00      4050.00
75%             48.60          18.70             213.00      4775.00
max             59.60          21.50             231.00      6300.00


In [3]:
# Langkah 3: Histogram + KDE untuk Flipper Length
plt.figure(figsize=(8, 5))
sns.histplot(data=df, x='flipper_length_mm', kde=True, bins=20, color='teal')
plt.title('Distribusi Panjang Sirip (Flipper Length) Pinguin')
plt.xlabel('Panjang Sirip (mm)')
plt.ylabel('Frekuensi')
plt.grid(True, alpha=0.3)
plt.show()

### Insight Grafik 1 - Histogram & KDE
Distribusi panjang sirip pinguin memperlihatkan pola bimodal (mempunyai dua puncak sebaran) di kisaran 190 mm dan 215 mm. Hal ini menandakan adanya kelompok populasi spesies pinguin yang secara fisik terbagi menjadi kelompok sirip lebih pendek dan kelompok sirip lebih panjang.

In [4]:
# Langkah 4: Boxplot per Spesies untuk Berat Tubuh (Body Mass)
plt.figure(figsize=(8, 5))
sns.boxplot(data=df, x='species', y='body_mass_g', palette='Set2')
plt.title('Perbandingan Berat Tubuh (Body Mass) per Spesies Pinguin')
plt.xlabel('Spesies')
plt.ylabel('Berat Tubuh (g)')
plt.grid(True, alpha=0.3)
plt.show()

### Insight Grafik 2 - Boxplot
Pinguin spesies **Gentoo** memiliki berat tubuh yang jauh lebih besar (rata-rata di atas 5000g) dibandingkan spesies **Adelie** dan **Chinstrap** yang memiliki berat rata-rata di kisaran 3700g. Spesies Chinstrap memiliki variasi berat tubuh paling rapat (kecil).

In [5]:
# Langkah 5: Matriks Korelasi Pearson Fitur Numerik
numeric_cols = df.select_dtypes(include=[np.number])
corr_matrix = numeric_cols.corr()

plt.figure(figsize=(7, 5))
sns.heatmap(corr_matrix, annot=True, cmap='coolwarm', fmt=".2f", linewidths=0.5)
plt.title('Matriks Korelasi Pearson Fitur Numerik Pinguin')
plt.show()

In [6]:
# Langkah 6: Scatter Plot antara Panjang Paruh vs Panjang Sirip
plt.figure(figsize=(8, 6))
sns.scatterplot(data=df, x='bill_length_mm', y='flipper_length_mm', hue='species', style='species', s=70)
plt.title('Hubungan Panjang Paruh vs Panjang Sirip Pinguin')
plt.xlabel('Panjang Paruh (mm)')
plt.ylabel('Panjang Sirip (mm)')
plt.grid(True, alpha=0.3)
plt.show()

### Insight Grafik 3 - Heatmap Korelasi & Scatter Plot
1. Berdasarkan Heatmap, terdapat korelasi positif yang sangat kuat antara panjang sirip (`flipper_length_mm`) dan berat badan (`body_mass_g`) yaitu sebesar **0.87**.
2. Pada Scatter Plot, kelompok spesies terpisah secara visual dengan jelas. Spesies Adelie mengelompok di area paruh pendek dan sirip pendek, spesies Gentoo mengelompok di paruh sedang dan sirip panjang, sedangkan spesies Chinstrap mengelompok di paruh panjang namun sirip pendek.

## Kesimpulan
Pada pertemuan keempat ini, saya mempelajari cara membuat visualisasi statistik menggunakan pustaka Matplotlib dan Seaborn:
1. Membuat Histogram dengan kurva KDE (`histplot(kde=True)`) untuk mengidentifikasi bentuk sebaran data numerik tunggal.
2. Membuat Boxplot (`boxplot()`) untuk mendeteksi ringkasan 5 angka (Q1, median, Q3, min, max) serta mendeteksi pencilan data secara visual per grup kategori.
3. Menghitung korelasi Pearson menggunakan `.corr()` dan memvisualisasikannya dengan Heatmap (`heatmap()`) untuk mengukur kekuatan hubungan linier antar fitur numerik.
4. Membuat Scatter Plot (`scatterplot()`) untuk memetakan hubungan dua variabel numerik kontinu serta membagi kluster berdasarkan kategori warna (*hue*).

**Temuan Utama**: Sebaran panjang sirip pinguin bersifat bimodal karena pencampuran karakteristik fisik antar spesies. Pinguin Gentoo adalah spesies terbesar, dan terdapat korelasi linear yang kuat antara panjang sirip dan berat badan pinguin.

**Keterbatasan/Pertanyaan**: Bagaimana cara merangkum berbagai grafik mandiri ini ke dalam satu tampilan dashboard statis yang rapi dan terpadu untuk presentasi?